docs/notebooks/model_config.pct.py

```diff
@@ -0,0 +1,209 @@
+#
+# Let's see this in action. We will re-use the `GPR` model we have created above and use the same additional arguments. As you can see, you retain all the flexibility but can skip working with the interfaces if you know them well already.
+
+# %%
+model_config = {
+    "model": gpflow_model,
+    "model_args": {
+        "num_kernel_samples": 100,
+    },
+    "optimizer": gpflow.optimizers.Scipy(),
+    "optimizer_args": {
+        "minimize_args": {"options": dict(maxiter=100)},
+    },
+}
+
+# %% [markdown]
+# Next you simply pass the configuration dictionary to the `optimize` function and `BayesianOptimizer` will sort out which model and optimizer wrapper needs to be used to build a valid model.
+
+# %%
+bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+result = bo.optimize(2, initial_data, model_config)
+
+
+# %% [markdown]
+# ## Using configuration dictionaries for setting up experiments
+#
+# Another use case is in setting up experiments, where it becomes easier to benchmark Bayesian optimization algorithms. The advantage is that we can easily change the models and set up any argument for them from one condition to another - only object with the experiment specification needs to change (`experiment_conditions` below), while the rest of the code for executing experiments can stay the same. 
+
+# %%
+from copy import deepcopy
+
+from trieste.models.gpflux import DeepGaussianProcess, build_vanilla_deep_gp
+
+
+def build_gp_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    kernel = gpflow.kernels.Matern52(variance=variance, lengthscales=[0.2, 0.2])
+    model = GPR(data.astuple(), kernel, noise_variance=1e-5)
+    return model
+
+
+def build_dgp_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    dgp = build_vanilla_deep_gp(data.query_points, num_layers=2, num_inducing=100)
+    dgp.f_layers[-1].kernel.kernel.variance.assign(variance)
+    dgp.f_layers[-1].mean_function = gpflow.mean_functions.Constant()
+    dgp.likelihood_layer.likelihood.variance.assign(1e-5)
+    return dgp
+
+
+def run_experiment(model_config):
+    bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+    result = bo.optimize(2, initial_data, model_config)
+    return result
+
+
+# configuration shared by all experiments, this is modified by each experiment condition
+basic_config = {
+    "model": gpflow_model,
+    "model_args": {
+        "num_kernel_samples": 100,
+    },
+    "optimizer": gpflow.optimizers.Scipy(),
+    "optimizer_args": {
+        "minimize_args": {"options": dict(maxiter=100)},
+    },
+}
+
+# here we specify our experiments
+experiment_conditions = [
+    {"model_args": {"num_kernel_samples": 500}},
+    {"model_args": {"num_kernel_samples": 1000}},
+    {"model": build_dgp_model(initial_data)},
+]
+
+results = []
+for exp in experiment_conditions:
+    model_config = deepcopy(basic_config)
```

Comment:
Should we provide a `combine_dict_configs` function that merges two dict configs? This would simplify this code and also handle properly merging nested configs (e.g. if one of the experiments tried to override maxiter).

```
def merge_dict_configs(config1: ModelDictConfig, config2: ModelDictConfig) -> ModelDictConfig:
   ...
```

Reply:
this is really an illustration only, I don't want to say this is the best way to do it - I'll add a comment reflecting this